AmazonChimeSDK/AmazonChimeSDKTests/device/DefaultDeviceControllerTests.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  DefaultDeviceControllerTests.swift
+//  AmazonChimeSDK
+//
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0
+//
+
+@testable import AmazonChimeSDK
+import AmazonChimeSDKMedia
+import AVFoundation
+import Mockingbird
+import XCTest
+
+class DefaultDeviceControllerTests: XCTestCase {
+    var audioSessionMock: AudioSessionMock!
+    var videoClientControllerMock: VideoClientControllerMock!
+    var loggerMock: LoggerMock!
+    var defaultDeviceController: DefaultDeviceController!
+
+    override func setUp() {
+        videoClientControllerMock = mock(VideoClientController.self)
+        loggerMock = mock(Logger.self)
+        audioSessionMock = mock(AudioSession.self)
+        defaultDeviceController = DefaultDeviceController(audioSession: audioSessionMock,
+                                                          videoClientController: videoClientControllerMock,
+                                                          logger: loggerMock)
+    }
+
+    func testListAudioDevices() {
+        let availableInputs = AVAudioSession.sharedInstance().availableInputs
```

Comment:
Can we create these ourselves instead of getting them from AVAudioSession?
In case they change in the future. Same for `testChooseAudioDevice_nonSpeaker`

Reply:
I was going that route but unfortunately I wasn't able to create an `AVAudioSessionPortDescription` object that's compatible with `MediaDevice.fromAVSessionPort(port: AVAudioSessionPortDescription)` which requires non empty port, while `AVAudioSessionPortDescription` is basically read-only.

It's still achievable to mock `AVAudioSessionPortDescription` as long as we create out own protocol and let `AVAudioSessionPortDescription` confirm to it. IMO the benefit (correct stubbing in UT) is not worth the effort and might not be the correct fix. The root cause is the assumption we currently make: there is always an input device otherwise SDK will crash as we insert "Built-in Speaker" device to position 1. Another cause is we're hard coding the input to be of `AVAudioSessionPortDescription` type which could be different on macOS and needs to be abstracted if later we want to make the SDK more generic.

I discussed with Weicheng about my concern on our assumption. We might want to address it for the default audio device API Nick is working on.